app/controllers/api/search_controller.rb

```diff
@@ -1,27 +1,37 @@
 class Api::SearchController < Api::ApiController
+  RESULTS_PER_PAGE = 25
+
   skip_authorization_check only: [:search, :suggestions]
 
   def search
     @current_page = [params[:page].to_i, 1].max
     @query = params[:q]
-    @filters = params[:filters]&.permit(:author, :thread, :subforum)&.to_h
+    @filters = params[:filters]&.permit(:author)&.to_h
```

Comment:
As far as I could tell, there wasn't actually a code path on the client that let us filter search results by thread or subforum, so I didn't duplicate the functionality.

You can autocomplete subforums and threads in the search box, but selecting them from the dropdown takes you to the appropriate subforum or thread, not to the search page.